source/slang/slang-compiler.cpp

```diff
@@ -196,6 +196,16 @@ namespace Slang
     // EntryPoint
     //
 
+    static const Guid IID_IEntryPoint = SLANG_UUID_IEntryPoint;
```

Comment:
This might be a little rad - but if our API is a C++ interface (that may be C compatible in the future with some code gen/COM ABIness), perhaps we should remove the macro stuff around guids, and say have a static member function in the interface that can return the interface. Doing so would mean we can always do (say) IEntryPoint::getInterfaceGuid(); . Doing as a function means that all this still works as a header (the interface can just return a static local variable with a ref to that).

It also means we don't have to define static local variables in the impl files that use the interfaces. 

The style means template mechanisms can be empowered to do more useful work too.

Reply:
Yeah, it could be good, so long as the functions are header-only (so they don't put C++ symbols into our DLL interface)